pycalphad/codegen/callables.py

```diff
@@ -200,8 +210,14 @@ def build_phase_records(dbf, comps, phases, conds, models, output='GM',
         callables = build_callables(dbf, comps, phases, models,
                                     parameter_symbols=parameters.keys(), output=output,
                                     additional_statevars=state_variables,
-                                    build_gradients=build_gradients,
-                                    build_hessians=build_hessians)
+                                    build_gradients=False,
+                                    build_hessians=False)
+    # XXX: Temporary; do not merge
+    formulacallables = build_callables(dbf, comps, phases, models,
+                                       parameter_symbols=parameters.keys(), output='G',
+                                       additional_statevars=state_variables,
+                                       build_gradients=build_gradients,
+                                       build_hessians=build_hessians)
 
```

Comment:
`build_phase_records` looks a lot better, but a few things:

1. Would it be possible/better to have the solver and phase records just use `output=G` from `equilibrium`? There's only two calls to `PhaseRecord.obj()` (see below). Can we simply replace them by dividing `PhaseRecord.formulamole_obj` by `sum(PhaseRecord.formulamole_obj)` and/or `CompsetState.energy/SystemState.phase_compositions[idx]` (for the latter)?

```
pycalphad/core/composition_set.pyx
53:        self.phase_record.obj(self._energy_2d_view, self.dof)

pycalphad/core/minimizer.pyx
891:                compset.phase_record.obj(phase_energies_per_mole_atoms[idx, :], x)
```

2. This is somewhat inefficient for `calculate` (or any general application of `build_callables` because it forces the derivatives to be calculated for the energy even if they are not useful. Fixing this may require changing `PhaseRecord` though and may be synergistic with item (1).

3. Having `build_gradients` and `build_hessians` set to `False` never allows non-`G` derivatives to be built. This is not something used, but technically a regression.

Reply:
I agree, but this is a little complicated to unwind because it isn't just `PhaseRecord.obj()` but also `PhaseRecord.obj_2d()` and `PhaseRecord.obj_parameters_2d()`

Reply:
This is actually a complicated API to change, I think. Reworking this to work in a more sensible way is probably a breaking change.

Reply:
The 2d versions are only used in`_compute_phase_values` and currently `calculate` will build its own phase records for that anyway.

Let's get this merged and clean up how we build and store phase records another day.